src/domain/data-decoder/v2/entities/contract.entity.ts

```diff
@@ -0,0 +1,32 @@
+import { z } from 'zod';
+import { buildPageSchema } from '@/domain/entities/schemas/page.schema.factory';
+import { AddressSchema } from '@/validation/entities/schemas/address.schema';
+import { HexSchema } from '@/validation/entities/schemas/hex.schema';
+
+export const ProjectSchema = z.object({
+  description: z.string(),
+  logo_file: z.string().url(),
+});
+
+export const AbiSchema = z.object({
+  // We could use abitype here, but we don't consume the ABI/it would increase entity complexity
+  abi_json: z.array(z.record(z.unknown())),
+  abi_hash: HexSchema,
+  modified: z.coerce.date(),
+});
+
+export const ContractSchema = z.object({
+  address: AddressSchema,
+  name: z.string(),
+  display_name: z.string().nullable(),
+  chain_id: z.number().transform(String),
+  project: ProjectSchema.nullable(),
+  abi: AbiSchema,
+  modified: z.coerce.date(),
+});
```

Comment:
@Uxio0 has confirmed that this (and decoded data) will be changed to camel case, hence leaving this as a draft.

Reply:
Addressed in https://github.com/safe-global/safe-client-gateway/pull/2313/commits/883ffdfdd51cde82f367d9eaf06fe7b1a3faeeb1.